src/main/java/bio/terra/app/controller/UpgradeApiController.java

```diff
@@ -0,0 +1,100 @@
+package bio.terra.app.controller;
+
+import bio.terra.controller.UpgradeApi;
+import bio.terra.model.JobModel;
+import bio.terra.model.UpgradeModel;
+import bio.terra.service.dataset.AssetModelValidator;
+import bio.terra.service.dataset.DatasetRequestValidator;
+import bio.terra.service.dataset.IngestRequestValidator;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.iam.AuthenticatedUserRequestFactory;
+import bio.terra.service.iam.PolicyMemberValidator;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshot.SnapshotRequestValidator;
+import bio.terra.service.upgrade.UpgradeService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.Optional;
+
+import static bio.terra.app.utils.ControllerUtils.jobToResponse;
```

Comment:
I think we should avoid static imports like this, except in cases like hamcrest matchers and asserts in tests.